nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/common/AbstractMQTTProcessor.java

```diff
@@ -384,16 +326,59 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
             onTrigger(context, session);
             session.commitAsync();
         } catch (final Throwable t) {
-            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            getLogger().error("{} failed to process due to {}; rolling back session", this, t);
             session.rollback(true);
             throw t;
         }
     }
 
     public abstract void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException;
 
-    protected boolean isConnected(){
+    protected boolean isConnected() {
         return (mqttClient != null && mqttClient.isConnected());
     }
 
+    protected MqttClientProperties getMqttClientProperties(final ProcessContext context) {
+        final MqttClientProperties clientProperties = new MqttClientProperties();
+
+        try {
+            clientProperties.setBrokerUri(new URI(context.getProperty(PROP_BROKER_URI).evaluateAttributeExpressions().getValue()));
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
The error should be changed to an `IllegalArgumentException` and should include a message.
```suggestion
            throw new IllegalArgumentException("Invalid Broker URI", e);
```

Reply:
Changed.